subsys/net/lib/coap/coap.c

```diff
@@ -808,94 +832,134 @@ struct coap_observer *coap_find_observer_by_addr(
 	return NULL;
 }
 
-u8_t *coap_packet_get_payload(struct coap_packet *cpkt, u16_t *len)
+int coap_packet_append_payload_marker(struct coap_packet *cpkt)
 {
-	u8_t *appdata = cpkt->pkt->frags->data;
-	u16_t appdatalen = cpkt->pkt->frags->len;
+	return net_pkt_append_u8(cpkt->pkt, COAP_MARKER) ? 0 : -EINVAL;
+}
 
-	if (!cpkt || !len) {
-		return NULL;
-	}
+int coap_packet_append_payload(struct coap_packet *cpkt, u8_t *payload,
+			       u16_t payload_len)
+{
+	bool status;
 
-	if (!cpkt->start) {
-		if (appdatalen + 1 >= cpkt->total_size) {
-			return NULL;
-		}
+	status = net_pkt_append_all(cpkt->pkt, payload_len, payload, K_FOREVER);
```

Comment:
Please use some timeout value for allocation. Either introduce a timeout to coap context, or use some define at start of the file like "#define ALLOC_TIMEOUT xxx"